tests/text/test_tabstops.py

```diff
@@ -47,9 +47,31 @@ def it_raises_on_indexed_access_when_empty(self):
         with pytest.raises(IndexError):
             tab_stops[0]
 
+    def it_can_add_a_tab_stop(self, add_tab_fixture):
+        tab_stops, new_pos, new_len, expected_xml = add_tab_fixture
+        tab = tab_stops.add_tab(new_pos)
+        assert len(tab_stops) == new_len
+        assert isinstance(tab, TabStop)
+        # assert tab_stops.???? == expected_xml
+
```

Comment:
Okay, so first thing is we don't want to use len() here, because that couples this test to the __len__() implementation, which already has its own test. The good news is we don't need it. So that line goes.

Instead, we can move the expected_xml line up there. Here's that line, then I'll explain why it works:

```python
assert tab_stops._element.xml == expected_xml
```

So the secret sauce here is that built into BaseOxmlElement is a property called `.xml` which returns the XML for that element. It's actually very handy for just taking a look at what an element's subtree looks like using a print() statement, but its core use is for just this, comparing against an expected XML snippet to make sure an operation produced the right XML. TabStops._element is the w:pPr element, so this should work just fine.

A fine point is I would use 'position' instead of 'pos'. Since .add_tab(position) is an API call, the parameter name is ours to define, and I'm thinking 'position' for that job; no reason to abbreviate and the full word is less ambiguous for an end-user learning the library. The name of the attribute `w:pos`, is *not* ours to define, and by convention we use a variable of that name to mean that attribute. So keeping them distinct makes clear to the reader we're not talking about an XML attribute here, just a float that says where we want the tab to be placed.

I would also avoid the 'new_' prefix since there's no other position to distinguish it from. It's generally good form if the main call in the test looks just like the API documentation, so in this case:

```python
tab_stop = tab_stops.add_tab_stop(position)
```

Which brings me to the method name. I think we want .add_tab_stop() here rather than add tab.